webgpu.h

```diff
@@ -65,6 +66,7 @@ typedef struct WGPUComputePassEncoderImpl* WGPUComputePassEncoder;
 typedef struct WGPUComputePipelineImpl* WGPUComputePipeline;
 typedef struct WGPUDeviceImpl* WGPUDevice;
 typedef struct WGPUFenceImpl* WGPUFence;
+typedef struct WGPUInstanceImpl* WGPUInstance;
```

Comment:
What's the benefit of having the separate `Instance` struct on native vs. just creating the adapter (without the adapter being attached to anything)?

Reply:
Some reasons off the top of my head:

- provides a place to map adapter ids to adapter objects
- provides a place for an event loop (however that ends up working)

Reply:
The Instance is the global object through which wgpu is used, and avoids the need for global static variables (unfortunately we have differing designs here). Global variables like these are useful for:

 - Dependency injection before the implementation does anything.
 - Caching queried adapters etc.
 - Setting global options (like enable validation, or start capture) that are sometimes needed by backends when we create adapters. (could be in descriptors but we don't want to expose that in webgpu.h, these are dawn-specific concerns)

Reply:
Oh, sorry for the race conditions